semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TextDocumentOps.scala

```diff
@@ -605,6 +604,15 @@ trait TextDocumentOps { self: SemanticdbOps =>
             }
 
             gtree match {
+              case v: g.ValDef
+                  if v.rhs.tpe.typeSymbol.isCaseClass && v.rhs.isInstanceOf[g.Apply] && v.rhs
+                    .asInstanceOf[g.Apply]
+                    .fun
+                    .symbol
+                    .isCaseApplyOrUnapply =>
+                if (gtree.pos != null && gtree.pos.isRange) {
```

Comment:
That might be one of the first conditions in the case's if above. Let's try to move easier checks above.

Reply:
Conditions changed.

Reply:
Let's move `gtree.pos != null` and `gtree.pos.isRange` above also - might even be one of the first condition. You should take look at what changed in the PR for `classOf`